pkgs/by-name/li/linux-pam/package.nix

```diff
@@ -2,97 +2,116 @@
-  autoreconfHook269,
-  pkg-config-unwrapped,
+  cmake,
+  meson,
+  pkg-config,
+  systemdLibs,
+  docbook5,
+  libxslt,
+  libxml2,
+  w3m-batch,
+  findXMLCatalogs,
+  docbook_xsl_ns,
+  nix-update-script,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "linux-pam";
-  version = "1.6.1";
+  version = "1.7.1";
 
-  src = fetchurl {
-    url = "https://github.com/linux-pam/linux-pam/releases/download/v${version}/Linux-PAM-${version}.tar.xz";
-    hash = "sha256-+JI8dAFZBS1xnb/CovgZQtaN00/K9hxwagLJuA/u744=";
+  src = fetchFromGitHub {
+    owner = "linux-pam";
+    repo = "linux-pam";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-kANcwxifQz2tYPSrSBSFiYNTm51Gr10L/zroCqm8ZHQ=";
   };
 
-  patches = [
-    ./suid-wrapper-path.patch
-    # required for fixing CVE-2025-6020
-    (fetchpatch {
-      url = "https://github.com/linux-pam/linux-pam/commit/10b80543807e3fc5af5f8bcfd8bb6e219bb3cecc.patch";
-      hash = "sha256-VS3D3wUbDxDXRriIuEvvgeZixzDA58EfiLygfFeisGg=";
-    })
-    # Manually cherry-picked from 475bd60c552b98c7eddb3270b0b4196847c0072e
-    ./CVE-2025-6020.patch
-  ];
-
-  # Case-insensitivity workaround for https://github.com/linux-pam/linux-pam/issues/569
-  postPatch =
-    lib.optionalString (stdenv.buildPlatform.isDarwin && stdenv.buildPlatform != stdenv.hostPlatform)
-      ''
-        rm CHANGELOG
-        touch ChangeLog
-      '';
+  # patching unix_chkpwd is required as the nix store entry does not have the necessary bits
+  postPatch = ''
+    substituteInPlace modules/module-meson.build \
+      --replace-fail "sbindir / 'unix_chkpwd'" "'/run/wrappers/bin/unix_chkpwd'"
+  '';
 
   outputs = [
     "out"
     "doc"
-    "man" # "modules"
+    "man"
+    # "modules"
   ];
 
   depsBuildBuild = [ buildPackages.stdenv.cc ];
-  # autoreconfHook269 is needed for `suid-wrapper-path.patch` above.
-  # pkg-config-unwrapped is needed for `AC_CHECK_LIB` and `AC_SEARCH_LIBS`
   nativeBuildInputs = [
+    cmake
     flex
-    autoreconfHook269
-    pkg-config-unwrapped
-  ]
-  ++ lib.optional stdenv.buildPlatform.isDarwin gettext;
+    meson
+    ninja
+    pkg-config
+    gettext
+
+    libxslt
+    libxml2
+    w3m-batch
+    findXMLCatalogs
+    docbook_xsl_ns
+    docbook5
+  ];
 
   buildInputs = [
     db4
     libxcrypt
   ]
-  ++ lib.optional stdenv.buildPlatform.isLinux audit;
+  ++ lib.optionals stdenv.buildPlatform.isLinux [
+    audit
+    systemdLibs
+  ];
 
   enableParallelBuilding = true;
 
-  configureFlags = [
-    "--includedir=${placeholder "out"}/include/security"
-    "--enable-sconfigdir=/etc/security"
-    # The module is deprecated. We re-enable it explicitly until NixOS
-    # module stops using it.
-    "--enable-lastlog"
+  mesonAutoFeatures = "auto";
+  mesonFlags = [
+    (lib.mesonEnable "logind" stdenv.buildPlatform.isLinux)
+    (lib.mesonEnable "audit" stdenv.buildPlatform.isLinux)
+    (lib.mesonEnable "pam_lastlog" true) # TODO: switch to pam_lastlog2
```

Comment:
Should we file a ticket for this?

Reply:
I opened https://github.com/NixOS/nixpkgs/issues/429075